lib/internal/Magento/Framework/Composer/ComposerInformation.php

```diff
@@ -148,7 +148,6 @@ public function getRequiredExtensions()
             /** @var CompletePackageInterface $package */
             foreach ($this->getLocker()->getLockedRepository()->getPackages() as $package) {
                 $requires = array_keys($package->getRequires());
-                $requires = array_merge($requires, array_keys($package->getDevRequires()));
                 $allPlatformReqs = array_merge($allPlatformReqs, $requires);
```

Comment:
Why do change this instead of `because isMagentoRoot works incorrectly for installation`?

Reply:
It collects all packages from composer.lock file and collects all require _**and require-dev**_ dependencies. But during composer install - require-dev dependencies are getting from composer.json file only. So fix is correct.

Reply:
@ihor-sviziev your statement does not answer my question.

> The root issue: method $this->isMagentoRoot() returns FALSE for current Magento installation, because its regular expression /magento\/magento2...?/ doesn't match magento/project-community-edition, that comes from metapackage.

Quote from the issue. Collected package information must be different depending whether it is a dev Magento installation or Composer-based.